source/ShipyardPanel.cpp

```diff
@@ -166,31 +179,44 @@ int ShipyardPanel::DrawDetails(const Point &center)
 {
 	string selectedItem = "No Ship Selected";
 	const Font &font = FontSet::Get(14);
-	const Color &bright = *GameData::Colors().Get("bright");
-	const Color &dim = *GameData::Colors().Get("medium");
-	const Sprite *collapsedArrow = SpriteSet::Get("ui/collapsed");
 
 	int heightOffset = 20;
 
 	if(selectedShip)
 	{
-		shipInfo.Update(*selectedShip, player, collapsed.count("description"));
+		const bool isShipyardShip = shipyard.Has(selectedShip);
+		// Ships in stock have no outfits, so get an appropriate version.
+		const Ship *infoShip = selectedShip;
+		// If the ship is license restricted, check to see if there's one in stock.
+		if(!isShipyardShip || LicenseCost(&selectedShip->Attributes()) < 0)
+			infoShip = player.StockShip(selectedShip);
+
+		shipInfo.Update(*infoShip, player, !isShipyardShip || collapsed.count("description"));
 		selectedItem = selectedShip->DisplayModelName();
 
-		const Sprite *background = SpriteSet::Get("ui/shipyard selected");
-		const Sprite *shipSprite = selectedShip->GetSprite();
-		float spriteScale = shipSprite
-			? min(1.f, (INFOBAR_WIDTH - 60.f) / max(shipSprite->Width(), shipSprite->Height()))
-			: 1.f;
+		// Find the old description zone and remove it.
```

Comment:
this is for variants? I'm not sure whats going on here

Reply:
Well, the code about that is for checking to see if the ship for sale is a generic store version or whether it's a stripped down version that the player sold to the store.  In the latter case it uses the special stripped down ship we made earlier for displaying ship details.

The code below that got shuffled around a bit to put variable declarations closer to where variables got used.  I also shifted the display of a couple of things a bit because they were a bit off (the ship name, the description zone).  And renamed a few variables because they'd been abbreviated.

Reply:
mb add a comment explaining what the code is doing?
I'm still confused about removing part of the description

Reply:
It's removing the description clickzone.  That's what lets you collapse and uncollapse the description text.  It needs to recalculate where the zone is, so it deletes the zone, does the calculation, then adds it back.

Reply:
I'm not sure why we have to do that tho, seems weird and idk if its the right way to do it

Reply:
You know, that's a good point.  I was mostly just following the existing code, but maybe I can just update the rectangle instead of deleting/adding it.